File: Cores/DotNet/Optimization/RemoteFunctions/RealRemoteFunction.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace OSOL.Extremum.Cores.DotNet.Optimization.RemoteFunctions
{
    public class RealRemoteFunction: RemoteFunction<double>
    {
                
        public RealRemoteFunction(string json, int port, string field) : base(json, port, field)
        {
            
        }

        public sealed override double Calculate(Dictionary<string, double> values)
        {
            string url = $"process_request?field={this.Field}";
            url += values.Select(kvp => $"&{kvp.Key}={kvp.Value}").Aggregate("", (p1, p2) => p1 + p2);
            var request = Client.DownloadString(url);
            
            return Convert.ToDouble(request);
        }
    }
}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Optimization/RemoteFunctions/RealRemoteFunction.scala
package OSOL.Extremum.Cores.JVM.Optimization.RemoteFunctions

import com.github.kevinsawicki.http.HttpRequest
import scala.collection.JavaConverters

class RealRemoteFunction(override val json: String, override val port: java.lang.Integer, override val field: String)
  extends RemoteFunction[java.lang.Double](json, port, field) {

  final override def apply(values: Map[String, java.lang.Double]): java.lang.Double = {
    val params = Map("field" -> this.field) ++ values.map { case (k, v) => (k, v.toString) }
    val request = HttpRequest.get(s"http://localhost:${this.port}/process_request", JavaConverters.mapAsJavaMap(params), true)

    java.lang.Double.parseDouble(request.body())
  }

}
